virtualizarr/xarray.py

```diff
@@ -438,3 +439,53 @@ def to_kerchunk(
             return None
         else:
             raise ValueError(f"Unrecognized output format: {format}")
+
+    def rename_paths(
+        self,
+        new: str | Callable[[str], str],
+    ) -> xr.Dataset:
+        """
+        Rename paths to chunks in every ManifestArray in this dataset.
+
+        Accepts either a string, in which case this new path will be used for all chunks, or
+        a function which accepts the old path and returns the new path.
+
+        Parameters
+        ----------
+        new
+            New path to use for all chunks, either as a string, or as a function which accepts and returns strings.
+
+        Returns
+        -------
+        Dataset
+
+        Examples
+        --------
+        Rename paths to reflect moving the referenced files from local storage to an S3 bucket.
+
+        >>> def local_to_s3_url(old_local_path: str) -> str:
+        ...     from pathlib import Path
+        ...
+        ...     new_s3_bucket_url = "http://s3.amazonaws.com/my_bucket/"
+        ...
+        ...     filename = Path(old_local_path).name
+        ...     return str(new_s3_bucket_url / filename)
+
+        >>> ds.virtualize.rename_paths(local_to_s3_url)
+
+        See Also
+        --------
+        ManifestArray.rename_paths
+        ChunkManifest.rename_paths
+        """
+
+        def _rename_manifestarray(marr: ManifestArray) -> ManifestArray:
+            return marr.rename_paths(new=new)
+
+        def _rename_dataarray(da: xr.DataArray) -> xr.DataArray:
+            if isinstance(da.data, ManifestArray):
+                return xr.apply_ufunc(_rename_manifestarray, da)
+            else:
+                return da
+
+        return self.ds.map(_rename_dataarray)
```

Comment:
Need to change this because `xr.apply_ufunc` craps out when it sees that the `ManifestArrays` have .chunks. Same problem really as #114, but we can just work around it here.

Reply:
Got around this in https://github.com/zarr-developers/VirtualiZarr/pull/152/commits/e099e9891509cb6d7b39dfc5a4f4d787c8f861d5